File: gvpr/Python3/GvprParserBase.py
from antlr4 import *
from gvprLexer import gvprLexer;

class GvprParserBase(Parser):

    def IsSemiRequired(self) -> bool:
        c = self._input.LT(-1)
        d = self._input.LT(1)
        return c.type != gvprLexer.CCBC

    def IsSemiNotRequired(self) -> bool:
        c = self._input.LT(-1)
        d = self._input.LT(1)
        return c.type == gvprLexer.CCBC
        

File: gvpr/TypeScript/GvprParserBase.ts
import {Parser, Lexer, Token, TokenStream, ParserRuleContext} from "antlr4";
import gvprParser from './gvprParser';
//import ParametersAndQualifiersContext from './CPP14Parser';

export default abstract class GvprParserBase extends Parser {

    constructor(input: TokenStream) {
        super(input);
    }

    IsSemiRequired() : boolean {
        var c = this._input.LT(-1);
        var d = this._input.LT(1);
        return c.type !== gvprParser.CCBC;
    }

    IsSemiNotRequired() : boolean {
        var c = this._input.LT(-1);
        var d = this._input.LT(1);
        return c.type === gvprParser.CCBC;
    }
}
